### PR TITLE
Fix single label multi obj class mapping

### LINKED ISSUE #386: Fix how the mapping to a single label works
Currently it works incorrectly, just losing the information.

### PR DESCRIPTION
**Pull request by abramov-oleg**

### CODE REVIEW

File: savant/deepstream/nvinfer/element_config.py [hunk 1]
```diff
@@ -32,7 +33,9 @@
 from savant.remote_file import process_remote
 from savant.utils.logging import get_logger
 
-__all__ = ['nvinfer_element_configurator']
+__all__ = ['nvinfer_element_configurator', 'MERGED_CLASSES']
+
+MERGED_CLASSES = defaultdict(dict)
```
Comment (bwsw) [inline savant/deepstream/nvinfer/element_config.py:38]: @abramov-oleg maybe use per-model scopped mapping, not global? It may introduce collisions.

Comment (abramov-oleg) [reply savant/deepstream/nvinfer/element_config.py:38]: Yes, I already use a two-level dict with the first level being tied to the element name. So, for example, multiple detector units based on the same detector model will not have any configuration collisions.

File: savant/deepstream/nvinfer/element_config.py [hunk 2]
```diff
@@ -411,14 +414,36 @@ def process(self, msg, kwargs):
...
-        object_labels = {
-            obj.class_id: obj.label for obj in model_config.output.get('objects')
-        }
-    model_uid = register_model_objects(
-        element_config.name, object_labels, RegistrationPolicy.Override
-    )
+    if issubclass(model_type, (ObjectModel, ComplexModel)):
+        output_objects = model_config.output.get('objects', [])
+        if output_objects:
+            logger.debug('Registering output objects for the model ...')
```
Comment (denisvmedyantsev) [inline savant/deepstream/nvinfer/element_config.py:420]: Try to name the model here, in logger.debug.

File: savant/deepstream/buffer_processor.py [hunk 3]
```diff
@@ -694,6 +695,19 @@ def prepare_element_output(self, element: PipelineElement, buffer: Gst.Buffer):
                                 obj_label = build_model_object_key(
                                     element.name, obj.label
                                 )
+                                obj_cls_id = MERGED_CLASSES[element.name].get(
```
Comment (denisvmedyantsev) [inline savant/deepstream/buffer_processor.py:698]: Are you sure there is no option for element.name not to be represented in MERGED_CLASSES?

**Review by denisvmedyantsev — APPROVED** [review]

File: savant/deepstream/buffer_processor.py (hunk 3 above)
Comment (abramov-oleg) [reply savant/deepstream/buffer_processor.py:698]: Should be ok, as the `MERGED_CLASSES` is a defaultdict

File: savant/deepstream/nvinfer/element_config.py (hunk 2 above)
Comment (abramov-oleg) [reply savant/deepstream/nvinfer/element_config.py:420]: It's already included in all messages in this function through the logging adapter defined above.

**Pull request merged**